### PR TITLE
Lg/anon

### PR DESCRIPTION
**Pull request by lukasgoetzweiss**

### CODE REVIEW

File: docs/guides/advanced-experimentation/anonymous-explainer.md
Comment (tbuffington7) [inline docs/guides/advanced-experimentation/anonymous-explainer.md:41]: Should this be a numbered step above? Specifically, it sounds like there is a step before step 3 that is "For each User ID, identify the first record with a non-NULL Anonymous ID and exclude all other records."

I think we should clarify the ordering of the filtering too. For example, do we filter out duplicated rows for each User ID _before_  we filter out mixed anonymous IDs? I'm wondering if there's a case where this filtering step removes the record that would have made an Anonmous ID mixed.

File: docs/guides/advanced-experimentation/anonymous-explainer.md
Comment (tbuffington7) [inline docs/guides/advanced-experimentation/anonymous-explainer.md:70]: Should this have an `OR ta.valid_from IS NULL`? Thinking about the case where an anonymous ID is the first one for a given `user_id`, so the lag function would return `NULL`. Ditto for `valid_to`

File: docs/guides/advanced-experimentation/anonymous-explainer.md
Comment (tbuffington7) [inline docs/guides/advanced-experimentation/anonymous-explainer.md:82]: This is true, but one might think "why not group the anonymous ids you know belong to the same user, even if we can't account for all of them?" It might be worth acknowledging the projected ID approach, even if it falls under the "Using your own existing attribution" approach

**Review by tbuffington7 — COMMENTED** [review]
Looks good overall and I left a few comments. I'd like to spend more time understanding this better before approving, but if others approve, feel free to merge

File: docs/guides/advanced-experimentation/anonymous-explainer.md
Comment (lukasgoetzweiss) [reply docs/guides/advanced-experimentation/anonymous-explainer.md:41]: IMO it's more "how step 3 works" as opposed to a separate step

I don't think we dedupe records for User ID, just for anon ID. If user 1 has cookie A and cookie B, each with their own treatment, neither cookie A nor cookie B would be filtered out

File: docs/guides/advanced-experimentation/anonymous-explainer.md
Comment (lukasgoetzweiss) [reply docs/guides/advanced-experimentation/anonymous-explainer.md:70]: yep definitely

File: docs/guides/advanced-experimentation/anonymous-explainer.md
Comment (tbuffington7) [reply docs/guides/advanced-experimentation/anonymous-explainer.md:41]: I think I'm confused then. This line makes it sound like we ARE deduplicating records for user id?

> Note that in step 3 we need to make sure each User ID is associated with only one Anonymous ID 

I'm struggling to reconcile that line with this scenario:

> If user 1 has cookie A and cookie B, each with their own treatment, neither cookie A nor cookie B would be filtered out

Doesn't this mean that a purchase from User 1 could be double-counted across cookie A or B? I feel like I'm missing something subtle.

File: docs/guides/advanced-experimentation/anonymous-explainer.md
Comment (lukasgoetzweiss) [reply docs/guides/advanced-experimentation/anonymous-explainer.md]: added a small further justification. i agree we shouldn't shy away from the projected ID thing, but we do mention the BYO approach right below, so hopefully the reader will know they can keep using their own approach if they disagree with ours

File: docs/guides/advanced-experimentation/anonymous-explainer.md
Comment (lukasgoetzweiss) [reply docs/guides/advanced-experimentation/anonymous-explainer.md:41]: Oh yeah I think I'm getting tripped up. You're right, each cookie can only have multiple users, but each user can only have one cookie

So I see your point: do we check if a cookie is associated with multiple variants before we remove other userIDs or after

@jonathan-fulton probably can speak more to that part

Comment (tbuffington7) [reply docs/guides/advanced-experimentation/anonymous-explainer.md:41]: Hmm, I don't have a strong opinion, but it feels a little more intuitive to me to use a user's first non-mixed cookie. My understanding is that our approach follows from the assumption that these cookie-randomized experiments are generally independent across sessions. So, if a user has a non-mixed cookie that isn't impacted by a previous mixed cookie, it seems like we might as well include it.

Comment (jonathan-fulton) [reply docs/guides/advanced-experimentation/anonymous-explainer.md:41]: So after writing the SQL, it's a bit tricky to do the first non-mixed cookie.  Right now in the code we're just going to pick the first cookie a user is associated with.

**Review by jonathan-fulton — APPROVED** [review]

**Pull request merged**
